File: app/src/main/java/towhid/icurious/webview/MainActivity.java
package towhid.icurious.webview;

import android.content.Context;
import android.net.ConnectivityManager;
import android.os.Bundle;
import android.os.Handler;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.webkit.WebView;
import android.webkit.WebViewClient;
import android.widget.Toast;


public class MainActivity extends AppCompatActivity {

    private WebView mWebView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mWebView = findViewById(R.id.webView);
        if (isOnline()) initWebView();
        viewSplash();
    }

    private void initWebView() {
        mWebView.getSettings().setJavaScriptEnabled(true);

        // Stop local links and redirects from opening in browser instead of WebView
        mWebView.setWebViewClient(new WebViewClient() {
            @Override
            public boolean shouldOverrideUrlLoading(WebView view, String url) {
                view.loadUrl(url);
                return true;
            }

            @Override
            public void onPageFinished(WebView view, String url) {
                findViewById(R.id.progressBar).setVisibility(View.INVISIBLE);
            }
        });

        mWebView.loadUrl(getString(R.string.url));
    }

    private void viewSplash() {
        if (getSupportActionBar() != null) getSupportActionBar().hide();
        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
//                getSupportActionBar().show();
                findViewById(R.id.splash).setVisibility(View.GONE);
            }
        }, 3000);
    }

    @Override
    public void onBackPressed() {
        if (mWebView.canGoBack()) mWebView.goBack();
        else super.onBackPressed();
    }

    public boolean isOnline() {
        ConnectivityManager cm = (ConnectivityManager) getSystemService(Context.CONNECTIVITY_SERVICE);
        boolean connected = cm != null && cm.getActiveNetworkInfo() != null && cm.getActiveNetworkInfo().isConnectedOrConnecting();

        if (!connected) Toast.makeText(this,
                "Internet Connection Required",
                Toast.LENGTH_LONG).show();
        return connected;
    }
}

File: app/src/main/java/towhid/icurious/webview/MainActivity.kt
package towhid.icurious.webview

import android.os.Bundle
import android.view.View
import android.view.ViewGroup
import android.view.WindowManager
import android.webkit.WebView
import android.webkit.WebViewClient
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*


class MainActivity : AppCompatActivity() {

    private val chromeClient: VideoEnabledWebChromeClient by lazy {
        object : VideoEnabledWebChromeClient(
                nonVideoLayout, // Your own view, read class comments
                videoLayout as ViewGroup, // Your own view, read class comments
                layoutInflater.inflate(R.layout.view_loading_video, null), // Your own view, read class comments
                webView) {
            override fun onProgressChanged(view: WebView, progress: Int) {
                // Subscribe to standard events, such as onProgressChanged()...
            }
        }.also {
            it.toggledFullscreenCallback = { fullscreen ->
                // Your code to handle the full-screen change, for example showing and hiding the title bar. Example:
                val attrs = window.attributes
                if (fullscreen) {
                    attrs.flags = attrs.flags or WindowManager.LayoutParams.FLAG_FULLSCREEN
                    attrs.flags = attrs.flags or WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON
                } else {
                    attrs.flags = attrs.flags and WindowManager.LayoutParams.FLAG_FULLSCREEN.inv()
                    attrs.flags = attrs.flags and WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON.inv()
                }
                window.attributes = attrs
                if (android.os.Build.VERSION.SDK_INT >= 14)
                    window.decorView.systemUiVisibility = if (fullscreen) View.SYSTEM_UI_FLAG_LOW_PROFILE else View.SYSTEM_UI_FLAG_VISIBLE
            }
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        supportActionBar?.hide()

        webView?.apply {
            setWebChromeClient(chromeClient)
            webViewClient = object : WebViewClient() {
                override fun shouldOverrideUrlLoading(view: WebView, url: String) = true.also { view.loadUrl(url) }
            }
            loadUrl(savedInstanceState?.getString("url", null) ?: getString(R.string.url))
        }
    }

    override fun onBackPressed() {
        if (!chromeClient.onBackPressed())  // Notify the VideoEnabledWebChromeClient, and handle it ourselves if it doesn't handle it
            if (webView.canGoBack()) webView.goBack() else super.onBackPressed()
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        outState.putString("url", webView.url)
    }
}